src/climsight/climsight.py

```diff
@@ -14,6 +14,14 @@
    datefmt='%Y-%m-%d %H:%M:%S'
 )
 
+# Initialize rag database
+try:
+    from rag import initialize_rag_database
+    initialize_rag_database()
+except Exception as e:
+    logger.warning(f"RAG database initialization skipped or failed: {e}")
+
```

Comment:
If skipLLMCall is True, we should avoid any calls to the LLM. Move the initialization below (only if skipLLMCall is False). Also after reading of config, pass config to RAG inizialization to avoid global variables. Add variable if RAG was found/inizialized.